Source/WebKit/Shared/Cocoa/CoreIPCPresentationIntent.mm

```diff
@@ -0,0 +1,126 @@
+        m_parentIntent = makeUnique<CoreIPCPresentationIntent>(intent.parentIntent);
+
+    switch (m_intentKind) {
+    case NSPresentationIntentKindHeader:
+        m_headerLevel = intent.headerLevel;
+        break;
+    case NSPresentationIntentKindListItem:
+        m_ordinal = intent.ordinal;
+        break;
+    case NSPresentationIntentKindCodeBlock:
+        m_languageHint = { intent.languageHint };
+        break;
+    case NSPresentationIntentKindTable:
+        for (NSNumber *alignment in intent.columnAlignments)
+            m_columnAlignments.append(alignment.unsignedIntegerValue);
+        m_columnCount = intent.columnCount;
+        break;
+    case NSPresentationIntentKindTableRow:
+        m_row = intent.row;
+        break;
+    case NSPresentationIntentKindTableCell:
+        m_column = intent.column;
+        break;
+    case NSPresentationIntentKindParagraph:
+    case NSPresentationIntentKindOrderedList:
+    case NSPresentationIntentKindUnorderedList:
+    case NSPresentationIntentKindBlockQuote:
+    case NSPresentationIntentKindThematicBreak:
+    case NSPresentationIntentKindTableHeaderRow:
+        break;
+    }
+}
+
+RetainPtr<id> CoreIPCPresentationIntent::toID() const
+{
+    id parent = (m_parentIntent ? m_parentIntent->toID().get() : nil);
+    NSPresentationIntent *result = nil;
+    NSMutableArray<NSNumber *> *columnAlignments = nil;
+    switch (m_intentKind) {
+    case NSPresentationIntentKindParagraph:
+        result = [NSPresentationIntent paragraphIntentWithIdentity:m_identity nestedInsideIntent:parent];
+        break;
+    case NSPresentationIntentKindHeader:
+        result = [NSPresentationIntent headerIntentWithIdentity:m_identity level:m_headerLevel nestedInsideIntent:parent];
+        break;
+    case NSPresentationIntentKindOrderedList:
+        result = [NSPresentationIntent orderedListIntentWithIdentity:m_identity nestedInsideIntent:parent];
+        break;
+    case NSPresentationIntentKindUnorderedList:
+        result = [NSPresentationIntent unorderedListIntentWithIdentity:m_identity nestedInsideIntent:parent];
+        break;
+    case NSPresentationIntentKindListItem:
+        result = [NSPresentationIntent listItemIntentWithIdentity:m_identity ordinal:m_ordinal nestedInsideIntent:parent];
+        break;
+    case NSPresentationIntentKindCodeBlock:
+        result = [NSPresentationIntent codeBlockIntentWithIdentity:m_identity languageHint:m_languageHint nestedInsideIntent:parent];
+        break;
+    case NSPresentationIntentKindBlockQuote:
+        result = [NSPresentationIntent blockQuoteIntentWithIdentity:m_identity nestedInsideIntent:parent];
+        break;
+    case NSPresentationIntentKindThematicBreak:
+        result = [NSPresentationIntent thematicBreakIntentWithIdentity:m_identity nestedInsideIntent:parent];
+        break;
+    case NSPresentationIntentKindTable:
+        columnAlignments = [[NSMutableArray alloc] initWithCapacity:m_columnAlignments.size()];
```

Comment:
This is a memory leak.

Reply:
It's not, because of the `[columnAlignments release]` that was further down, but I've changed to use smart pointers to be consistent with The WK Way™.